### PR TITLE
test(progress-tracker): add accessibility tests

### PR DESCRIPTION
**Pull request by DlgSHi**
### Proposed behaviour
- Add `accessibility` Cypress tests for the `Progress-Tracker` component to use the new cypress-component-react framework for testing.

### Current behaviour
Old approach for accessibility tests

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

- [x] Run `npx cypress open --component` to check if there are newly added tests
- [x] Check if the `progress-tracker.test.js` file passed
- [x] Run `npx cypress run --component` to check none of the other *.test.js files have regressed
- [x] Run `npx cypress run --e2e` to check none of the feature files have regressed
- [x] Run `npm run build-storybook` -> `npx sb extract` -> `npx http-server storybook-static -p 9001` and run `npx cypress run --e2e` and check that `Progress Tracker` tests are not running twice.

### CODE REVIEW

File: src/components/progress-tracker/progress-tracker.test.js [hunk 1]
```diff
@@ -169,4 +181,135 @@ context("Tests for ProgressTracker component", () => {
       }
     );
   });
+
+  describe("should checkt accessibility tests for progress tracker component", () => {
```
Comment (ZhuoyuJin) [inline src/components/progress-tracker/progress-tracker.test.js:185]: A typo here"describe("should **check** accessibility tests for progress tracker component", () => {"

File: src/components/progress-tracker/progress-tracker.test.js [hunk 2]
```diff
@@ -169,4 +181,135 @@ context("Tests for ProgressTracker component", () => {
...
+      (labelsPosition) => {
+        CypressMountWithProviders(
+          <ProgressTrackerComponent labelsPosition={labelsPosition} />
+        );
+
+        cy.checkAccessibility();
+      }
+    );
+  });
 });
```
Comment (ZhuoyuJin) [inline src/components/progress-tracker/progress-tracker.test.js:315]: Do we not need to make some changes to stories.tsx file like putting "export const ProgressTrackerComponent" from the current test.js file into that tsx file?

**Review by ZhuoyuJin — CHANGES_REQUESTED** [review]

File: src/components/progress-tracker/progress-tracker.test.js [hunk 3]
```diff
@@ -28,15 +28,27 @@ context("Tests for ProgressTracker component", () => {
       "aria-valuetext",
     ])("when %s prop is passed", (propName) => {
       it(`verify that the ${propName} is set to cypress-standard`, () => {
-        const props = { [propName]: CHARACTERS.STANDARD };
+        let props;
+        if (
+          !propName.endsWith("now") &&
```
Comment (edleeks87) [inline src/components/progress-tracker/progress-tracker.test.js:33]: You're checking this condition twice might be worth storing it in a variable

File: src/components/progress-tracker/progress-tracker.test.js [hunk 4]
```diff
@@ -28,15 +28,27 @@ context("Tests for ProgressTracker component", () => {
...
+          !propName.endsWith("min") &&
+          !propName.endsWith("max")
+        ) {
+          props = { [propName]: CHARACTERS.STANDARD };
+        } else {
+          props = { [propName]: 50 };
```
Comment (edleeks87) [inline src/components/progress-tracker/progress-tracker.test.js:39]: You could store this magic number as a const as well

File: src/components/progress-tracker/progress-tracker.test.js (hunk 1 above)
Comment (edleeks87) [inline src/components/progress-tracker/progress-tracker.test.js:185]:
```suggestion
  describe("should check accessibility tests for progress tracker component", () => {
```

**Comment by divyajindel** [comment]
@DlgSHi I've observed "description" prop is not covered in your tests so could you please take a look.

File: src/components/progress-tracker/progress-tracker.test.js [hunk 5]
```diff
@@ -169,4 +181,135 @@ context("Tests for ProgressTracker component", () => {
...
+  describe("should checkt accessibility tests for progress tracker component", () => {
+    describe.each([
+      "aria-label",
+      "aria-describedby",
+      "aria-valuenow",
+      "aria-valuemin",
+      "aria-valuemax",
+      "aria-valuetext",
+    ])("when %s prop is passed", (propName) => {
+      it(`check the accessibility when the ${propName} is set to cypress-standard`, () => {
```
Comment (edleeks87) [inline src/components/progress-tracker/progress-tracker.test.js:194]:
```suggestion
      it("conforms to cypress-standard for accessibility", () => {
```

File: src/components/progress-tracker/progress-tracker.test.js (hunk 3 above)
Comment (edleeks87) [reply src/components/progress-tracker/progress-tracker.test.js:33]: As you've also done the same check below maybe move it to a function but I'm happy for you to just go with with just storing it in constants if you believe the below isn't worth the effort etc

```
// globally in file
const DEFAULT_PROP_VALUE = 50;

const checkPropName = (propName) => propName.endsWith("now") && propName.endsWith("min") && propName.endsWith("max");

const getProps = (propName, shouldBeDefault) => {
if (shouldBeDefault) {
  return { [propName]: DEFAULT_PROP_VALUE };
}
  return { [propName]: CHARACTERS.STANDARD };
}


// inside test cases 
const isNowMinOrMax = checkPropName(propName);

const props = getProps(propName, isNowMinOrMax);

const assertion = isNowMinOrMax ? DEFAULT_PROP_VALUE : CHARACTERS.STANDARD;

...
```

File: src/components/progress-tracker/progress-tracker.test.js [hunk 6]
```diff
@@ -169,4 +181,135 @@ context("Tests for ProgressTracker component", () => {
...
+    ])(
+      "should check the accessibility when component is rendered with %s size",
+      (size) => {
+        CypressMountWithProviders(<ProgressTrackerComponent size={size} />);
+
+        cy.checkAccessibility();
+      }
+    );
+
+    it.each([150, 350, 550])(
```
Comment (edleeks87) [inline src/components/progress-tracker/progress-tracker.test.js:225]: Could you not just pass these as strings with `px` and update the test description to be:
```
it.each(["150px", "350px", "550px"])(
"should check the accessibility when component is rendered %s length",
(length) => {
        CypressMountWithProviders(
          <ProgressTrackerComponent length={length} />
        );
```

File: src/components/progress-tracker/progress-tracker.test.js (hunk 2 above)
Comment (DlgSHi) [reply src/components/progress-tracker/progress-tracker.test.js:315]: this component is still not refactored to `ts` so there is no point of doing that

**Review by edleeks87 — APPROVED** [review]

**Comment by DlgSHi** [comment]
> @DlgSHi I've observed "description" prop is not covered in your tests so could you please take a look.

@divyajindel this is not a part of this task, but yes it needs to be added

**Review by ZhuoyuJin — DISMISSED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Pull request merged**